bpf/lib/icmp6.h

```diff
@@ -524,4 +524,42 @@ icmp6_host_handle(struct __ctx_buff *ctx, int l4_off, __s8 *ext_err, bool handle
 #endif /* ENABLE_HOST_FIREWALL */
 }
 
+static __always_inline
+bool icmp6_ndisc_validate(struct __ctx_buff *ctx, const struct ipv6hdr *ip6,
+			  union macaddr *iface_mac, const union macaddr *smac,
+			  union v6addr *sip, union v6addr *tip)
+{
+	__u8 nexthdr = ip6->nexthdr;
+	struct icmp6hdr *icmp;
+	int l3_off = (int)((__u8 *)ip6 - (__u8 *)ctx_data(ctx));
+	int l4_off = ipv6_hdrlen(ctx, &nexthdr);
+	struct ethhdr *eth = (void *)(long)ctx->data;
```

Comment:
nit for consistency and simplification:
```suggestion
	struct ethhdr *eth = ctx_data(ctx);
```

Reply:
Good catch. Fixed.